tools/cli/go-whisk-cli/commands/util.go

```diff
@@ -473,6 +473,26 @@ func addKeyValue(key string, value interface{}, keyValueArr whisk.KeyValueArr) (
     return append(keyValueArr, keyValue)
 }
 
+func findKeyValue(keyValueArr whisk.KeyValueArr, key string) (int) {
+    for i := 0; i < len(keyValueArr); i++ {
+        if strings.ToLower(keyValueArr[i].Key) == strings.ToLower(key) {
+            return i
+        }
+    }
+
+    return -1
+}
+
+func appendKeyValueArr(origKeyValueArr whisk.KeyValueArr, appKeyValueArr whisk.KeyValueArr) (whisk.KeyValueArr) {
+    for i := 0; i < len(appKeyValueArr); i++ {
+        if findKeyValue(origKeyValueArr, appKeyValueArr[i].Key) == -1 {
```

Comment:
This seems like a good function candidate to the whisk.KeyValueArr structure:
```
func (kvr KeyValueArr) AppendKeyValueArr(kvrToAdd KeyValueArr) KeyValueArr {
...
}
```
Minor nit... I suggest renaming the arg/var `fetch` to reflect its higher purpose of "preserve all existing annotations"